### ISSUE TITLE
Metadata version mismatch for module found version 4, expected 3

### ISSUE THREAD

**Issue by anathematic** [bugsnag/bugsnag-angular#12]
I'm having issues building my Angular 4.2 app with bugsnag-angular 2.x + bugsnag-js 4.6.x.
Looks like this is only occurring with the `-prod` or `-aot` flag.

```bash
ng build -prod
ERROR in Error: Metadata version mismatch for module PATH_TO_APP/node_modules/bugsnag-angular/dist/index.d.ts, found version 4, expected 3
    at StaticSymbolResolver.getModuleMetadata (PATH_TO_APP/node_modules/@angular/compiler/bundles/compiler.umd.js:25616:34)
    at StaticSymbolResolver._createSymbolsOf (PATH_TO_APP/node_modules/@angular/compiler/bundles/compiler.umd.js:25404:46)
    at StaticSymbolResolver.getSymbolsOf (PATH_TO_APP/node_modules/@angular/compiler/bundles/compiler.umd.js:25385:14)
    at PATH_TO_APP/node_modules/@angular/compiler/bundles/compiler.umd.js:24241:30
    at Array.forEach (<anonymous>)
    at extractProgramSymbols (PATH_TO_APP/node_modules/@angular/compiler/bundles/compiler.umd.js:24240:79)
    at AotCompiler.analyzeModulesAsync (PATH_TO_APP/node_modules/@angular/compiler/bundles/compiler.umd.js:23796:47)
    at CodeGenerator.codegen (PATH_TO_APP/node_modules/@angular/compiler-cli/src/codegen.js:32:14)
    at Function.NgTools_InternalApi_NG_2.codeGen (PATH_TO_APP/node_modules/@angular/compiler-cli/src/ngtools_api.js:73:30)
    at _donePromise.Promise.resolve.then (PATH_TO_APP/node_modules/@angular/cli/node_modules/@ngtools/webpack/src/plugin.js:391:44)
    at <anonymous>
```

Here's my code:

```typescript
// app.module.ts
import { BrowserModule } from '@angular/platform-browser';
import { NgModule } from '@angular/core';

// Import bugsnag-js and bugsnag-angular
import BugsnagErrorHandler from 'bugsnag-angular'
import bugsnag from 'bugsnag-js'

// configure Bugsnag ASAP, before any other imports
const bugsnagClient = bugsnag('API_KEY')

// create a factory which will return the bugsnag error handler
export function errorHandlerFactory() {
  return new BugsnagErrorHandler(bugsnagClient)
}

import { AppComponent } from './app.component';

@NgModule({
  declarations: [
    AppComponent
  ],
  imports: [
    BrowserModule
  ],
  providers: [],
  bootstrap: [AppComponent]
})
export class AppModule { }
```

I've wrapped this up in a [sample app](https://github.com/anathematic/angular-bugsnag-example) for easier debugging,  any pointers would be amazing!

**Comment by bengourley** [comment]
Hey @anathematic, thanks for getting in touch.

It looks like the build output from the angular v5 toolchain is not backwards compatible with v4 apps. I've shipped a prerelease that builds using the v4 toolchain, which appears to be forwards compatible and so works on both v4 and v5 apps.

Please can you `npm install bugsnag-angular@2.0.1-0` and let me know if this fixes your issue? I have already tried it on your sample app (thanks for providing that, super helpful!) and it worked there.

Cheers

**Comment by anathematic** [comment]
G'day @bengourley,
Thanks a bunch for the prompt reply, being new to Angular this was a little over my head.

I've just tested `2.0.1-0` and it worked beautifully, thanks so much!

**Comment by bengourley** [comment]
Great, this is now published as v2.0.1 🚀
